### PR TITLE
Quickfix for Spaces and Tabs in Keyword

### PR DESCRIPTION
**Pull request by cdietrich**
…458). Added a Quickfix that replaces spaces and tabs in keyword with _

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by svenefftinge** [comment]
Replacing with underscore is usually not what people want, but splitting a keyword into multiple ones.

**Comment by cdietrich** [comment]
i thought of that but that fix is not generally applicable (unassigned keywords)

**Comment by JanKoehnlein** [comment]
You mean assigned keywords?
I'd rather have QuickFix "a split into two keywords" for the cases where it is applicable (I guess that is unassigned keywords, correct me if I am wrong), and no fix for the other. An issue in both cases.

**Comment by cdietrich** [comment]
i mean the quickfix works for "my kw" but not for assign="my kw"

**Comment by cdietrich** [comment]
now i have a quickfix for the "simple" cases

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java [hunk 1]
```diff
@@ -164,6 +169,42 @@ public void apply(final EObject element, IModificationContext context) {
...
+	public void fixSpacesInKeyword(final Issue issue, IssueResolutionAcceptor acceptor) {
+		if (issue.getData().length > 0 && Boolean.valueOf(issue.getData()[0]))
+		acceptor.accept(issue, "Replace Keyword with multiple Keywords", "Splits a Keyword that contains spaces or tabs into multiple Keywords", NULL_QUICKFIX_IMAGE,
+				new ISemanticModification() {
+			@Override
+			public void apply(final EObject element, IModificationContext context) {
+				Keyword keyword = (Keyword) element;
+				EStructuralFeature eContainingFeature = keyword.eContainingFeature();
+				EObject eContainer = keyword.eContainer();
+				String[] parts = keyword.getValue().split("[\\t ]");
```
Comment (JanKoehnlein) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java]: How about
  ...split("\\s*");

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java [hunk 2]
```diff
@@ -164,6 +169,42 @@ public void apply(final EObject element, IModificationContext context) {
 					}
 				});
 	}
+	@Fix(SPACES_IN_KEYWORD)
+	public void fixSpacesInKeyword(final Issue issue, IssueResolutionAcceptor acceptor) {
+		if (issue.getData().length > 0 && Boolean.valueOf(issue.getData()[0]))
+		acceptor.accept(issue, "Replace Keyword with multiple Keywords", "Splits a Keyword that contains spaces or tabs into multiple Keywords", NULL_QUICKFIX_IMAGE,
+				new ISemanticModification() {
```
Comment (JanKoehnlein) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java]: Don't you think a textual modification would be simpler, also for the case where you have to insert perentheses?

**Comment by cdietrich** [comment]
so is it ok if we live without a quickfix?

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java (hunk 2 above)
Comment (cdietrich) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java]: but where to put the logic? inside the validator? calling xtextdocument.readonly?

**Comment by svenefftinge** [comment]
Yes, I agree. It doesn't seem to be worth the effort.

**Pull request closed without merging**